### PR TITLE
fix(ci): add sanity check to build.rs scripts, fixed "detected dubious ownership in repository" errors on `git`

### PR DESCRIPTION
**Pull request by jkomyno**
This PR checks that `git rev-parse HEAD`, the CLI command used at compilation time to store the hash in the Query Engine and Schema Engine, exits successfully.

It also complements https://github.com/prisma/prisma-engines/pull/4947 by running `git config --global --add safe.directory /root/build` when cross-building via `prismagraphql/build:*`. This avoids "fatal: detected dubious ownership in repository at /root/build" errors in Linux Musl ARM64 and Linux Static ARM64 architectures.

This was introduced after third-party clients notified us that `./query-engine --version` no longer displayed the version hash in Linux Alpine.

Fixes: https://github.com/prisma/prisma/issues/25020
Fixes: https://github.com/prisma/prisma/issues/24971

### CODE REVIEW

**Review by SevInf — APPROVED** [review]

File: .github/workflows/build-engines.yml [hunk 1]
```diff
@@ -69,6 +69,7 @@ jobs:
           (
             github.event.pull_request.author_association == 'OWNER' ||
             github.event.pull_request.author_association == 'MEMBER' ||
+            github.event.pull_request.author_association == 'CONTRIBUTOR' ||
```
Comment (aqrln) [inline .github/workflows/build-engines.yml:72]: why is this change necessary in this PR?

File: schema-engine/cli/build.rs [hunk 2]
```diff
@@ -2,6 +2,16 @@ use std::process::Command;
...
+        panic!(
+            "Failed to get git commit hash.\nstderr: \n{}\nstdout {}\n",
+            String::from_utf8(output.stderr).unwrap_or_default(),
+            String::from_utf8(output.stdout).unwrap_or_default(),
+        );
+    }
+
     let git_hash = String::from_utf8(output.stdout).unwrap();
     println!("cargo:rustc-env=GIT_HASH={git_hash}");
 }
```
Comment (aqrln) [inline schema-engine/cli/build.rs:17]: nit: would be nicer to have this defined once in some kind of build utils crate that these three crates will depend on as a build dependency

File: .github/workflows/build-engines.yml (hunk 1 above)
Comment (jkomyno) [reply .github/workflows/build-engines.yml:72]: Yes, otherwise the Linux Build step would've been skipped. My user role is "CONTRIBUTOR", not "COLLABORATOR", according to the action's logs

File: schema-engine/cli/build.rs (hunk 2 above)
Comment (jkomyno) [reply schema-engine/cli/build.rs:17]: Yes, the whole `store_git_commit_hash` could be abstracted

Comment (jkomyno) [reply schema-engine/cli/build.rs:17]: Nit postponed to https://github.com/prisma/team-orm/issues/1261
